rust/core-lib/src/editor.rs

```diff
@@ -302,9 +304,14 @@ impl<W: Write + Send + 'static> Editor<W> {
         }
     }
 
+    fn has_unsaved_changes(&self) -> bool {
```

Comment:
The general style in xi is to update the state fully when things change, then have querying of the state as trivial as possible (which is what @cmyr is suggesting). But if it's cheap to compute and simpler, I don't have a big problem with doing more work on query.